### ISSUE TITLE
Crash on embedded SVGs in jruby, 1.6.7.2

### ISSUE THREAD

**Issue by toddgardner** [sparklemotion/nokogiri#1461]
Created a reproducer; this was occurring while trying to rearrange a site with an embedded SVG.

```
jruby-9.0.4.0 :001 > require 'nokogiri'
 => true
jruby-9.0.4.0 :002 > puts Nokogiri::VERSION
1.6.7.2
 => nil
jruby-9.0.4.0 :003 > Nokogiri::HTML::DocumentFragment.parse('<svg version="1.1" viewbox="0 0 1024 1024" x="0px" xml:space="preserve" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" y="0px"></svg>')
Java::JavaLang::NullPointerException: 
    from nokogiri.internals.NokogiriHelpers.renameNode(NokogiriHelpers.java:839)
    from nokogiri.XmlNode.relink_namespace(XmlNode.java:499)
    from nokogiri.XmlNodeSet.relink_namespace(XmlNodeSet.java:121)
    from nokogiri.XmlDocumentFragment.relink_namespace(XmlDocumentFragment.java:208)
    from nokogiri.XmlNode.adoptAs(XmlNode.java:1558)
    from nokogiri.XmlNode.add_child_node(XmlNode.java:1677)
    from nokogiri.XmlNode$INVOKER$i$1$0$add_child_node.call(XmlNode$INVOKER$i$1$0$add_child_node.gen)
    from org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:313)
    from org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:163)
    from org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:290)
    from org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:77)
    from org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:83)
    from org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:197)
    from org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:183)
    from org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:197)
    from org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:313)
... 182 levels...
    from org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:183)
    from org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:197)
    from org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:313)
    from org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:163)
    from Users.todd.$_dot_rvm.rubies.jruby_minus_9_dot_0_dot_4_dot_0.bin.jirb.invokeOther13:start(/Users/todd/.rvm/rubies/jruby-9.0.4.0/bin/jirb)
    from Users.todd.$_dot_rvm.rubies.jruby_minus_9_dot_0_dot_4_dot_0.bin.jirb.RUBY$script(/Users/todd/.rvm/rubies/jruby-9.0.4.0/bin/jirb:13)
    from java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:636)
    from org.jruby.ir.Compiler$1.load(Compiler.java:111)
    from org.jruby.Ruby.runScript(Ruby.java:821)
    from org.jruby.Ruby.runScript(Ruby.java:813)
    from org.jruby.Ruby.runNormally(Ruby.java:751)
    from org.jruby.Ruby.runFromMain(Ruby.java:573)
    from org.jruby.Main.doRunFromMain(Main.java:409)
    from org.jruby.Main.internalRun(Main.java:304)
    from org.jruby.Main.run(Main.java:233)
    from org.jruby.Main.main(Main.java:200)
```

**Comment by jvshahid** [comment]
worth noting that the example you provided throws an exception on master but with a different stack trace (possibly related to the refactor in https://github.com/sparklemotion/nokogiri/commit/9caa43ecb3ea5a2ac4952db44b62183377b18796). I was able to reproduce the stack trace on the commit prior to the one mentioned before. Either way, I confirm this is a bug.

**Comment by flavorjones** [comment]
@kares Here's another one you may want to look at.

**Comment by jmirmina** [comment]
We are running into this issue as well. If it helps, in the original example, `xml:space="preserve"` seems to be whats causing the error. When I remove it, it seems to parse fine.

**Comment by flavorjones** [comment]
Apologies for the long reply time; this was fixed in v1.6.8, likely as part of @jvshahid's overhaul of namespace handling.
